### DISCUSSION TITLE
adgh 不支持h2c？

### DISCUSSION THREAD

**Discussion by Potterli20** [AdguardTeam/AdGuardHome#3983] [Q&A]
adgh 不支持h2c？我网站用了http2，但是采用h2c
![image_2021-12-22_18-05-58](https://user-images.githubusercontent.com/28734349/147090575-46d4a4a3-369a-4d58-a7c5-76ba3155b6ee.png)

在官方文档中，为HTTP/2协议定义了两个版本：h2 和 h2c：

h2版本的协议是建立在TLS层之上的HTTP/2协议，这个标志被用在TLS应用层协议协商（TLS-ALPN）域和任何其它的TLS之上的HTTP/2协议。
h2c版本是建立在明文的TCP之上的HTTP/2协议，这个标志被用在HTTP/1.1的升级协议头域和其它任何直接在TCP层之上的HTTP/2协议。

**Comment by ameshkov** [comment]
AGH 确实不支持 H2C。如果你不用 TLS 就需要用 HTTP/1.1.

**Reply by Potterli20** [reply]
问题是我另一个项目在用h2c，要依赖h2c的，所以这种没有办法调整？

**Reply by Potterli20** [reply]
我用 curl --http2-prior-knowledge是这样用·的

**Reply by ameshkov** [reply]
@Potterli20 你只有两个办法，第一个是用 HTTP/1.1，第二个是用支持 h2c 的代理。

**Reply by Potterli20** [reply]
> @Potterli20 你只有两个办法，第一个是用 HTTP/1.1，第二个是用支持 h2c 的代理。

其实我是想采用本地的127.0.0.1,我已经有第二的方法。
